src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorRenameFeature.ts

```diff
@@ -0,0 +1,97 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { getRazorDocumentUri, isRazorCSharpFile } from './RazorConventions';
+import { RazorDocumentManager } from './RazorDocumentManager';
+import { RazorDocumentSynchronizer } from './RazorDocumentSynchronizer';
+import { RazorLanguage } from './RazorLanguage';
+import { RazorLanguageServiceClient } from './RazorLanguageServiceClient';
+import { RazorLogger } from './RazorLogger';
+import { RazorRenameProvider } from './RazorRenameProvider';
+import { LanguageKind } from './RPC/LanguageKind';
+
+export class RazorRenameFeature {
+    private readonly renameProvider: RazorRenameProvider;
+
+    constructor(
+        documentSynchronizer: RazorDocumentSynchronizer,
+        documentManager: RazorDocumentManager,
+        private readonly serviceClient: RazorLanguageServiceClient,
+        logger: RazorLogger) {
+            this.renameProvider = new RazorRenameProvider(
+                documentSynchronizer,
+                documentManager,
+                serviceClient,
+                logger);
+    }
+
+    public register() {
+        const registrations = [
+            vscode.languages.registerRenameProvider(RazorLanguage.id, this.renameProvider),
+
+            // Register a command that will be called whenever an edit happens to any CSharp file.
+            // It takes in the edits that Omnisharp is planning to do and re-map any Razor edits to the
+            // original Razor documents.
+            vscode.commands.registerCommand(
```

Comment:
Does this currently account for your O# design that's in the works? Wondering if you planned to come back after and fix this up depending on how that turned out. Doesn't matter either way, just curious.

Reply:
Not completely no. I plan to revisit this based on how that turns out. I expect the changes to be minimal though.